arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -460,23 +478,34 @@ arangodb::Result RocksDBReplicationContext::dumpKeys(
     }
   }
 
-  auto cb = [&](LocalDocumentId const& documentId, VPackSlice slice) {
-    TRI_voc_rid_t revisionId = 0;
-    VPackSlice key;
-    transaction::helpers::extractKeyAndRevFromDocument(slice, key, revisionId);
-
-    TRI_ASSERT(key.isString());
+  auto cb = [&](rocksdb::Slice const& rocksKey, rocksdb::Slice const& rocksValue) {
+    TRI_voc_rid_t docRev;
+    if(rocksValue.size()
+       >= sizeof(std::uint64_t) /*doc id*/
+       +  sizeof(TRI_voc_rid_t) /*revision id*/
+    ) {
```

Comment:
Can we extract this logic and put it in a helper method on `RocksDBValue` the same way we have e.g. `RocksDBValue::documentId` below? So we could have `RocksDBValue::hadRevisionId` to check and `RocksDBValue::revisionId` to extract.